src/modules/dashboard/components/Dialogs/EditExpenditureDialog/ChangedMultiple/ChangedMultiple.tsx

```diff
@@ -0,0 +1,145 @@
+import { FieldArray } from 'formik';
+import { isNil } from 'lodash';
+import React, { Fragment } from 'react';
+import { defineMessages, FormattedMessage, useIntl } from 'react-intl';
+
+import Button from '~core/Button';
+import { FormSection } from '~core/Fields';
+import { Colony, LoggedInUser } from '~data/index';
+import { ValuesType } from '~pages/ExpenditurePage/types';
+import { Staged } from '~dashboard/ExpenditurePage/Staged/types';
+
+import ChangeItem from './ChangeItem';
+import styles from './ChangedMultiple.css';
+
+export const MSG = defineMessages({
+  newChange: {
+    id: 'dashboard.EditExpenditureDialog.ChangedMultiple.newChange',
+    defaultMessage: '{count}: {changeType}',
+  },
+  discard: {
+    id: 'dashboard.EditExpenditureDialog.ChangedMultiple.discard',
+    defaultMessage: 'Discard',
+  },
+  from: {
+    id: 'dashboard.EditExpenditureDialog.ChangedMultiple.from',
+    defaultMessage: 'From',
+  },
+  changeTo: {
+    id: 'dashboard.EditExpenditureDialog.ChangedMultiple.changeTo',
+    defaultMessage: 'Change to',
+  },
+  recipient: {
+    id: 'dashboard.EditExpenditureDialog.ChangedMultiple.recipient',
+    defaultMessage: 'Recipient',
+  },
+});
+
+const displayName = 'dashboard.EditExpenditureDialog.ChangedMultiple';
+const skip = ['id', 'claimDate', 'isExpanded', 'created', 'removed'];
+
+type NewValuesType = {
+  id: string;
+  key: string;
+  value?:
+    | ValuesType['recipients']
+    | string
+    | Staged
+    | Pick<
+        LoggedInUser,
+        'walletAddress' | 'balance' | 'username' | 'ethereal' | 'networkId'
+      >;
+}[];
+
+interface Props {
+  newValues?: NewValuesType;
+  colony: Colony;
+  oldValues: ValuesType;
+}
+
+const ChangedMultiple = ({ newValues, oldValues, colony }: Props) => {
+  const { formatMessage } = useIntl();
+
+  return (
+    <>
+      {newValues?.map((newValue) => (
+        <Fragment key={newValue.id}>
+          {Array.isArray(newValue.value) &&
+            newValue.value?.map((changeItem, index) => {
+              const oldItem = oldValues[newValue.key]?.find(
+                (item) => item?.id === changeItem?.id,
+              );
+              return (
+                <FieldArray
```

Comment:
This component is very "heavy". I think it would be great to have a child component with part of the logic/html separated there